lib/tui/widget.ml

```diff
@@ -105,8 +105,10 @@ let fmt_file ~max_name_len (tree : Fs.tree) =
       match Lazy.force contents with
       | Fs.Text _ -> file_char ^ " " ^ pad name
       | Fs.Binary -> bin_char ^ " " ^ pad name)
-  | Dir (name, [||]) -> empty_dir_char ^ " " ^ pad name
-  | Dir (name, _) -> dir_char ^ " " ^ pad name
+  | Dir (name, children) -> (
+      match Lazy.force children with
+      | [||] -> empty_dir_char ^ " " ^ pad name
+      | _ -> dir_char ^ " " ^ pad name)
```

Comment:
Same comment about `lazy` pattern. We can still keep the `match` inside to make patterns slightly more manageable:

```suggestion
  | Dir (name, lazy children) -> (
      match children with
      | [||] -> empty_dir_char ^ " " ^ pad name
      | _ -> dir_char ^ " " ^ pad name)
```